arch/armv7a/memcpy.S

```diff
@@ -0,0 +1,361 @@
+#define DST r0
+#define SRC r1
+#define LEN r8
+#define DST_RET r12
+
+.arm
+.syntax unified
+
+.text
+.align 4
+
+/* void *memcpy(void *dst, const void *src, size_t l) */
+.globl memcpy
+.type memcpy, %function
+memcpy:
+	str r8, [sp, #-8]!
+	mov r8, r2
+	cmp LEN, #64
+	mov DST_RET, DST /* preserve return value */
+
+	bhs .LblkCopy
+
+	/* less than 64 bytes - always copy as if block was always unaligned */
+
+.Ltail63Unaligned:
+	/* unaligned copy, 0-63 bytes */
+
+	/* r3 = LEN / 4 */
+	movs r3, LEN, lsr #2
+	beq .Ltail63Un0
+
+.Ltail63Un4:
+	ldr r2, [SRC], #4
+	str r2, [DST], #4
```

Comment:
To be able to copy by 4 bytes **both** `SRC` and `DST` need to be 4-byte aligned